mathesar/api/db/permissions/schema.py

```diff
@@ -0,0 +1,21 @@
+from django.db.models import Q
+from rest_access_policy import AccessPolicy
+
+from mathesar.models.users import Role
+
+
+class SchemaAccessPolicy(AccessPolicy):
+
+    @classmethod
+    def scope_queryset(cls, request, qs):
+        if not request.user.is_superuser:
+            allowed_roles = (Role.MANAGER.value,)
+
+            if request.method.lower() == 'get':
+                allowed_roles = allowed_roles + (Role.EDITOR.value, Role.VIEWER.value)
+            permissible_databaserole_filter = (
```

Comment:
Can we call this `database_role` for consistency? Same with `schema_role` below.